shared/constants/types/more.js

```diff
@@ -85,3 +85,9 @@ export type DumbComponentMap<C: Component<*, *, *>> = {
     [key: string]: PropsOf<C>,
   },
 }
+
+export type LooseRecord<T> = T & {
```

Comment:
This seems to work better than Immutable's typedef of `Record`. It allows you to access things through destructuring and `.` syntax (i.e. state.foo)